src/controllers/time.rs

```diff
@@ -132,14 +132,14 @@ impl TimeController {
             state.world.player.advance_wrapping(dt * speed, state.world.size);
 
             // Update resource
-            state.world.player.resource.update();
+            state.world.player.gun.cool_down();
```

Comment:
We should pass the `dt` variable here to the `cool_down` function. See my comments to `cool_down` for context.